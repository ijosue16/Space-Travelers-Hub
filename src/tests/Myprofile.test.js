import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import MyProfile from '../components/myprofile';
import store from '../redux/configurestore';
import '@testing-library/jest-dom';

describe('tests for my profile', () => {
  test('test if my rockets is present in file and has no class', () => {
    render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );
    expect(screen.queryByText(/rockets reserved/i)).toBeNull();
    expect(screen.queryByText(/My rockets/i)).not.toHaveClass();
  });

  it('Myprofile renders correctlly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MyProfile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
