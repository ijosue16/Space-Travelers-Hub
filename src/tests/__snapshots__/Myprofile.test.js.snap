// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for my profile Myprofile renders correctlly 1`] = `
<div
  className="container-fluid d-block d-md-flex gap-2 "
>
  <div
    className="flex-fill"
  >
    <h4
      className=""
    >
      My rockets
    </h4>
    <ul
      className="list-group"
    >
      <li
        className="border-0"
      >
        No reserved rockets yet!!
      </li>
    </ul>
  </div>
  <div
    className="flex-fill"
  >
    <h4
      className=""
    >
      My missions
    </h4>
    <ul
      className="list-group"
    >
      <li
        className="border-0 list-unstyled"
      >
        No reserved missions yet!!
      </li>
    </ul>
  </div>
</div>
`;
